app/models/remote_authenticatable.rb

```diff
@@ -0,0 +1,119 @@
+module Devise
+  module Models
+    module RemoteAuthenticatable
+      extend ActiveSupport::Concern
+
+      #
+      # Here you do the request to the external webservice
+      #
+      # If the authentication is successful you should return
+      # a resource instance
+      #
+      # If the authentication fails you should return false
+      #
+      #def remote_authentication(authentication_hash)
+      def remote_authentication(params)
+        if Rails.application.secrets.cognito_aws_access_key_id.present? &&
+            Rails.application.secrets.cognito_aws_secret_access_key.present?
+          client = Aws::CognitoIdentityProvider::Client.new(
+            region: Rails.application.secrets.aws_region,
+            access_key_id: Rails.application.secrets.cognito_aws_access_key_id,
+            secret_access_key: Rails.application.secrets.cognito_aws_secret_access_key
+          )
+        else
+          client = Aws::CognitoIdentityProvider::Client.new
```

Comment:
Does this mean we're creating a new client on every authentication? Is there a way around this?